packages/components/popover/tests/popover.test.tsx

```diff
@@ -111,6 +111,119 @@ describe("<Popover />", () => {
     await waitFor(() => expect(footer).toBeVisible())
   })
 
+  test("should contextmenu props work", async () => {
+    const { user } = render(<PopoverExample trigger="contextmenu" />)
+
+    const triggerButton = await screen.findByRole("button", {
+      name: "Open Popover",
+    })
+
+    await user.pointer({ target: triggerButton, keys: "[MouseRight]" })
+
+    const header = await screen.findByText("Popover Header")
+    const closeButton = await screen.findByLabelText("Close popover")
+    const body = await screen.findByText("Popover Body")
+    const footer = await screen.findByText("Popover Footer")
+
+    await waitFor(() => expect(header).toBeVisible())
+    await waitFor(() => expect(closeButton).toBeVisible())
+    await waitFor(() => expect(body).toBeVisible())
+    await waitFor(() => expect(footer).toBeVisible())
+  })
+
+  test("should close with escape key", async () => {
+    const { user } = render(<PopoverExample />)
+
+    const triggerButton = await screen.findByRole("button", {
+      name: "Open Popover",
+    })
+
+    await user.click(triggerButton)
+
+    const header = await screen.findByText("Popover Header")
+    const closeButton = await screen.findByLabelText("Close popover")
+    const body = await screen.findByText("Popover Body")
+    const footer = await screen.findByText("Popover Footer")
+
+    await user.keyboard("{Escape}")
+
+    await waitFor(() => expect(header).not.toBeVisible())
+    await waitFor(() => expect(closeButton).not.toBeVisible())
+    await waitFor(() => expect(body).not.toBeVisible())
+    await waitFor(() => expect(footer).not.toBeVisible())
```

Comment:
I’m not sure if this will solve the issue, but how about trying to modify the code as follows? @Shunii85 
```suggestion
await waitFor(() => {
  expect(header).not.toBeVisible();
  expect(closeButton).not.toBeVisible();
  expect(body).not.toBeVisible();
  expect(footer).not.toBeVisible();
}, { timeout: 3000 });
```